File: feature-scanner-impl/src/main/java/com/example/scanner/presentation/view/ScannerMainView.java
package com.example.scanner.presentation.view;

import moxy.MvpView;
import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

public interface ScannerMainView extends MvpView {

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showScannerWork();
    @StateStrategyType(OneExecutionStateStrategy.class)
    void showScannerSuccess();
    @StateStrategyType(OneExecutionStateStrategy.class)
    void showBuyWork();
    @StateStrategyType(OneExecutionStateStrategy.class)
    void showBuySuccess();

}

File: feature-scanner-impl/src/main/java/com/example/scanner/presentation/view/ScannerMainView.kt
package com.example.scanner.presentation.view

import moxy.MvpView
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType
import moxy.viewstate.strategy.alias.OneExecution

@OneExecution
interface ScannerMainView : MvpView {
    fun showScannerWork()

    fun showScannerSuccess()

    fun showBuyWork()

    fun showBuySuccess()
}